tests/scripts/components-build-system.sh

```diff
@@ -11,21 +11,16 @@
 
 component_test_tf_psa_crypto_cmake_out_of_source () {
     msg "build: cmake tf-psa-crypto 'out-of-source' build"
-    TF_PSA_CRYPTO_ROOT_DIR="$PWD"
-    mkdir "$OUT_OF_SOURCE_DIR"
-    cd "$OUT_OF_SOURCE_DIR"
     # Note: Explicitly generate files as these are turned off in releases
     cmake -D CMAKE_BUILD_TYPE:String=Check -D GEN_FILES=ON "$TF_PSA_CRYPTO_ROOT_DIR"
     make
     msg "test: cmake tf-psa-crypto 'out-of-source' build"
     make test
-    cd "$TF_PSA_CRYPTO_ROOT_DIR"
-    rm -rf "$OUT_OF_SOURCE_DIR"
 }
 
 component_test_tf_psa_crypto_cmake_as_subdirectory () {
     msg "build: cmake 'as-subdirectory' build"
-    cd programs/test/cmake_subproject
+    cd  "$TF_PSA_CRYPTO_ROOT_DIR/programs/test/cmake_subproject"
```

Comment:
The CI is not happy with $TF_PSA_CRYPTO_ROOT_DIR. I am not sure why we need it in the first place. In TF-PSA-Crypto only context we do not need it. In mbedtls CI context I think `tests/scripts/all.sh` goes `cd tf-psa-crypto; ./tests/scripts/all.sh` thus it should be needed either. But I may miss something.

Reply:
The problem that I have ran into is how we identify tests that require out of source builds and tests that don't. The current solution at [line 410](https://github.com/Mbed-TLS/mbedtls-framework/blob/1797ef20b0c8aff96146b682b782fbc31fabb855/scripts/all-core.sh#L410) in the framework pull request just sets up an out of build directory regardless of the test.

Reply:
Ah yes I see, sorry for the noise. We need that for this very peculiar component but will not need it for "standard" ones.